File: data/readable_format/cpp/file1139.cpp
#include <iostream>
#include <vector>
#include <cstring>
#include <algorithm>

class VaryingString {
public:
    std::vector<char> chars;

    // Assignment from a C-style string
    VaryingString& operator=(const char* expr) {
        chars.assign(expr, expr + std::strlen(expr));
        return *this;
    }
};

// Function to check if two ranges of characters are not equal
bool notEqual(const std::vector<char>& vec, const char* str, size_t len) {
    return !std::equal(vec.begin(), vec.begin() + len, str);
}

int main() {
    // Simulating the equivalence of char_arb and char_elm
    // by using a single character array and manipulating pointers.
    char char_storage[10]; // Storage for both char_arb and char_elm
    char* char_arb[2] = {char_storage, char_storage + 5};
    char* char_elm = char_storage;

    // Copying strings into the shared storage
    std::strcpy(char_arb[0], "Hello");
    std::strcpy(char_arb[1], "World");

    // Creating VaryingString instances and assigning values
    VaryingString str_ara[2];
    str_ara[0] = char_arb[0];
    str_ara[1] = char_arb[1];

    // Performing the checks as in the original Fortran program
    if (notEqual(str_ara[0].chars, char_elm, 5)) return 1;
    if (notEqual(str_ara[1].chars, char_elm + 5, 5)) return 2;

    return 0;
}

File: data/readable_format/fortran/file1139.f90
module iso_varying_string
    implicit none
    integer, parameter :: GET_BUFFER_LEN = 256
    type varying_string
        character(LEN=1), dimension(:), allocatable :: chars
    end type varying_string
    interface assignment(=)
        module procedure op_assign_VS_CH
    end interface assignment(=)
contains
    elemental subroutine op_assign_VS_CH (var, expr)
        type(varying_string), intent(out) :: var
        character(LEN=*), intent(in) :: expr
        var = var_str(expr)
    end subroutine op_assign_VS_CH
    elemental function var_str (chr) result (string)
        character(LEN=*), intent(in) :: chr
        type(varying_string) :: string
        integer :: length
        integer :: i_char
        length = LEN(chr)
        ALLOCATE(string%chars(length))
        forall(i_char = 1:length)
            string%chars(i_char) = chr(i_char:i_char)
        end forall
    end function var_str
end module iso_varying_string

program VST_2
    use iso_varying_string
    implicit none
    CHARACTER(LEN=5) :: char_arb(2)
    CHARACTER(LEN=1) :: char_elm(10)
    equivalence (char_arb, char_elm)
    type(varying_string) :: str_ara(2)

    char_arb(1) = "Hello"
    char_arb(2) = "World"
    str_ara = char_arb
    
    if (any(str_ara(1)%chars(1:5) /= char_elm(1:5))) stop 1
    if (any(str_ara(2)%chars(1:5) /= char_elm(6:10))) stop 2

    print *, "All tests passed successfully."
end program VST_2